horovod/tensorflow/gradient_aggregation.py

```diff
@@ -205,7 +205,11 @@ def compute_gradients(self, grads, vars):
             allreduced_grads = tf.cond(
                 tf.equal(self.counter, self.backward_passes_per_step),
                 lambda: self._allreduce_grads_helper(grads, vars),
-                lambda: grads,
+                lambda: (
+                    [tf.convert_to_tensor(g) if g is not None and isinstance(g, tf.IndexedSlices) else g
```

Comment:
I proposed this to the user as a one-liner to make it easy to hotpatch, but I think we should clean it up in the real fix here.

Can we make this a tiny little helper function like:

```python
def _maybe_convert_grad(self, grad):
    if self.sparse_as_dense and isinstance(g, tf.IndexedSlices):
        return tf.convert_to_tensor(g)
    return g
```

We can then do 
```python
tf.cond(..., false=lambda: [self._maybe_convert_grad(g) for g in grads])
```
here, and replace lines 80-81 with
```python
grad = self._maybe_convert_grad(grad)
```


.